libr/search/aes-find.c

```diff
@@ -49,31 +53,51 @@ static bool aes128_key_test(const unsigned char *buf) {
 }
 
 R_API int r_search_aes_update(RSearch *s, ut64 from, const ut8 *buf, int len) {
-	int i, last = len - 20;
+	int i, t, last = len - AES128_MIN_SEARCH_LENGTH;
 	RListIter *iter;
 	RSearchKeyword *kw;
+	const int old_nhits = s->nhits;
 
 	r_list_foreach (s->kws, iter, kw) {
-		if (last > 0) {
+		if (last >= 0) {
 			for (i = 0; i < last; i++) {
 				if (aes128_key_test (buf + i)) {
 					kw->keyword_length = 16;
-					return r_search_hit_new (s, kw, from + i);
+					t = r_search_hit_new (s, kw, from + i);
+					i += AES128_MIN_SEARCH_LENGTH;
+					if (!t) {
+						return -1;
+					}
+					if (t > 1) {
+						return s->nhits - old_nhits;
+					}
 				}
-				if (len - i - 28 > 0) {
-					if (aes192_key_test (buf + i)) {
-						kw->keyword_length = 24;
-						return r_search_hit_new (s, kw, from + i);
+				if (len - i - AES192_MIN_SEARCH_LENGTH >= 0 && aes192_key_test (buf + i)) {
+					kw->keyword_length = 24;
```

Comment:
why do we test for `AES192_MIN_SEARCH_LENGTH` and then do `kw->keyword_length = 24`? Same in the other cases.

Reply:
A 192-bit AES key is 24 bytes so we use `kw->keyword_length = 24` as output to print the key enteriely.  AES192_MIN_SEARCH_LENGTH  give the minimum number of bytes you need to perform the search if it is less you cannot search.

Reply:
so I guess `AES192_MIN_SEARCH_LENGTH` should actually be named `AES192_KEY_LENGTH` and it should be used both in the line above and here.

Reply:
this ^

Reply:
@sylvainpelissier still this ^

Reply:
It was patch in a commit: https://github.com/sylvainpelissier/radare2/blob/02014abbb429bc660aa53842404299c15de888b0/libr/search/aes-find.c#L15

Reply:
I have added them and rebased.